### PR TITLE
Invalid query should result in HTTP 400

### PR DESCRIPTION
**Pull request by adamretter**
Closes https://github.com/eXist-db/exist/issues/1646

### CODE REVIEW

File: test/src/org/exist/http/RESTServiceTest.java [hunk 1]
```diff
@@ -33,6 +33,7 @@
 import java.net.URLEncoder;
 import javax.xml.parsers.ParserConfigurationException;
 
+import org.eclipse.jetty.http.HttpStatus;
```
Comment (dizzzz) [inline test/src/org/exist/http/RESTServiceTest.java]: better to use https://docs.oracle.com/javaee/6/api/javax/servlet/http/HttpServletResponse.html#SC_NOT_FOUND ?

Comment (adamretter) [reply test/src/org/exist/http/RESTServiceTest.java]: I don't see our dependency on Jetty going away, but I am happy to revise it. Do you wish me to do so?

File: src/org/exist/http/RESTServer.java [hunk 2]
```diff
@@ -1854,26 +1854,24 @@ private void writeXPathException(final HttpServletResponse response,
...
+            writer.write("<message>");
+            final String message = e.getMessage() == null ? e.toString() : e.getMessage();
+            writer.write(XMLUtil.encodeAttrMarkup(message));
+            writer.write("</message>");
+            if (query != null) {
+                writer.write("<query>");
+                writer.write(XMLUtil.encodeAttrMarkup(query));
+                writer.write("</query>");
+            }
+            writer.write("</exception>");
```
Comment (dizzzz) [inline src/org/exist/http/RESTServer.java]: Don't we need a flush() call here to be sure all data is sent to the http agent?

Comment (dizzzz) [reply src/org/exist/http/RESTServer.java]: answering myself: no not according to the servlet spec :-)

**Comment by adamretter** [comment]
@dizzzz I changed this back to 4.0.0 milestone because it is an API change.

**Comment by dizzzz** [comment]
ah ok, sorry.

**Pull request merged**